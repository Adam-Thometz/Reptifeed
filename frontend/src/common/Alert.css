.Alert {
  width: 25vw;
  margin: 5px auto;
  border: 1px solid #D5CE3B;
  color: #D5CE3B;
  padding: 1%;;
  border-radius: 25px;
}

.danger {
  background-color: rgb(138, 0, 0)
}

.success {
  background-color: green;
}