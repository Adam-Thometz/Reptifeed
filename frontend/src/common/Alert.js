import React from "react";

const Alert = ({ type = 'danger', messages = [] }) => {
  return (
    <div className={`Alert-${type}`}>
      {messages.map(m => (
        <p key={m}>{m}</p>
      ))}
    </div>
  );
};

export default Alert;