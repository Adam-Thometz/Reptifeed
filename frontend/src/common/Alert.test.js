import React from "react";
import { render } from '@testing-library/react'
import Alert from "./Alert";

test('it renders without crashing', () => {
  render(<Alert />);
});

test('it matches snapshot for danger', () => {
  let messages = ["NOOOOOOOO", "Why did you do that?????"];
  const { asFragment } = render(<Alert type="danger" messages={messages} />);
  expect(asFragment()).toMatchSnapshot();
});

test('it matches snapshot for success', () => {
  let messages = ["You did it!", "You have an incredible work ethic!"];
  const { asFragment } = render(<Alert type="success" messages={messages} />);
  expect(asFragment()).toMatchSnapshot();
});