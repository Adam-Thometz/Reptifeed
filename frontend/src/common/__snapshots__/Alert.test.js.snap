// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it matches snapshot for danger 1`] = `
<DocumentFragment>
  <div
    class="Alert-danger"
  >
    <p>
      NOOOOOOOO
    </p>
    <p>
      Why did you do that?????
    </p>
  </div>
</DocumentFragment>
`;

exports[`it matches snapshot for success 1`] = `
<DocumentFragment>
  <div
    class="Alert-success"
  >
    <p>
      You did it!
    </p>
    <p>
      You have an incredible work ethic!
    </p>
  </div>
</DocumentFragment>
`;
